providers/firetext/src/lib/firetext.provider.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  ChannelTypeEnum,
+  ISendMessageSuccessResponse,
+  ISmsOptions,
+  ISmsProvider,
+} from '@novu/stateless';
+
+if (!globalThis.fetch) {
+  // eslint-disable-next-line global-require
+  globalThis.fetch = require('node-fetch');
+}
+
+export class FiretextSmsProvider implements ISmsProvider {
+  id = 'firetext';
+  channelType = ChannelTypeEnum.SMS as ChannelTypeEnum.SMS;
+  private BASE_URL = 'https://www.firetext.co.uk/api/sendsms';
+
+  constructor(
+    private config: {
+      apiKey: string;
+      from: string;
+    }
+  ) {}
+
+  private parseResponse(body: string) {
+    const re = /^(\d+):(\d+)\s(.*)$/i;
+    const found = body.match(re);
+    const code = found?.[1] ?? 'Unknown status code';
+    const message = found?.[3] ?? 'Unknown status message';
+
+    return [code, message];
+  }
+
+  private parseHeaderId(headers: Headers) {
+    return headers.get('X-Message');
+  }
+
+  async sendMessage(
+    options: ISmsOptions
+  ): Promise<ISendMessageSuccessResponse> {
+    const baseMessage = {
+      apiKey: this.config.apiKey,
+      to: options.to,
+      from: this.config.from,
+      message: options.content,
+    };
+
+    const urlSearchParams = new URLSearchParams(baseMessage);
+    const url = new URL(this.BASE_URL);
+    url.search = urlSearchParams.toString();
+
+    const response = await fetch(url.toString());
+    const body = await response.text();
+    const [code, message] = this.parseResponse(body);
+
+    if (code !== '0') {
+      throw new Error(`${code}: ${message}`);
+    }
```

Comment:
Checking Firetext documentation they have several potential errors. Should we map them and having a dedicated enum for them to be able to control the errors and handle exceptions better?
https://www.firetext.co.uk/docs#sendingsms-xml

Reply:
I wasn't sure how errors are handled upstream from providers so I've opted to include all the error information in the error message I have available if not successful. Is there scope to handle errors differently based on error type?

My approach here was just to pass through errors without doing too much, but there are a few alternatives:

1. Create specific error types for each + an enum to map the errors. e.g. `new InsufficientCreditError('Insufficient credit')`
2. Create an enum that just explicitly lists the possible errors but continue to use the standard `Error`. 

I think (1) might be useful for error handling higher up, however (2) makes it more verbose but perhaps better documented.

Reply:
I personally tend to go with the first approach as it also provides enough self documenting too.
Anyway is something can be done in a future PR. 👍🏻